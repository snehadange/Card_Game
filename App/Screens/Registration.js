import React from 'react'
import {
    StyleSheet, View, Text, ImageBackground, Image, TextInput, ScrollView, SafeAreaView,
    Alert, ActivityIndicator
} from 'react-native';
import { IconsPath } from '../Assets';
import { TouchableOpacity } from 'react-native-gesture-handler';
import { Ionicons, MaterialIcons } from 'react-native-vector-icons';



export default class Registration extends React.Component {

    constructor() {
        super()
        this.state = {
            name: '',
            nameError: '',
            email: '',
            emailError: '',
            number: '',
            numberError: '',
            password: '',
            passwordError: 'dgh',
            icEye: 'visibility-off',
            showPassword: true,
        }

        this.baseurl = "http://192.168.1.102:8000/src/router"
    }



    validator = (type) => {

        let rjx = /^[a-zA-Z]*$/;
        let emailrjx = /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/;
        let numberrjx = /^[0-9]{10}$/;
        let passwordrjx = /^(?=.*[A-Za-z])(?=.*\d)[A-Za-z\d]{8,}$/;




        if (type == 'name') {
            if (rjx.test(this.state.name)) {
                this.setState({
                    nameError: ''
                })

            } else {
                this.setState({
                    nameError: 'Name must be in alphabets only'
                })

            }
        }

        else if (type == 'password') {
            if (passwordrjx.test(this.state.password)) {
                this.setState({
                    passwordError: ''
                })
            } else {
                this.setState({
                    passwordError: 'Password must be Minimum eight characters, at least one letter and one number '
                })
            }
        }


        else if (type == 'email') {
            if (emailrjx.test(this.state.email)) {
                this.setState({
                    emailError: ''
                })
            } else {
                this.setState({
                    emailError: 'This is not a valid email address'
                })
            }
        }

        else if (type == 'number') {
            if (numberrjx.test(this.state.number)) {
                this.setState({
                    numberError: ''
                })
            } else {
                this.setState({
                    numberError: 'Mobile number must be 10 digits'
                })
            }
        }

    }


    register = () => {
        const { navigate } = this.props.navigation

        let collections = {
            name: this.state.name,
            email: this.state.email,
            password: this.state.password,
            phoneNumber: this.state.number
        };
        let url = this.baseurl + '/registerUser';

        if (this.state.name == '') {
            alert("Name is required")

        } else if (this.state.number == '') {
            alert("Phone number is required")
        } else if (this.state.email == '') {
            alert("Email is required")
        } else if (this.state.password == '') {
            alert("password is required")
        }
        else {

            fetch(url,
                {
                    method: 'POST', // or 'PUT'

                    headers: {
                        'Accept': 'application/json',
                        'Content-Type': 'application/json'
                    },
                    body: JSON.stringify(collections),

                })
                .then(response => response.json())

                .then(data => {

                    console.log('Successfull:', data);

                    this.setState({
                        name: '',
                        email: '',
                        password: '',
                        number: ''
                    })




                    if (data.errors) {
                        alert("Data not inserted")

                        data.errors.map((err) => {

                            if (err.param == 'name') {
                                this.setState({
                                    nameError: err.msg
                                })
                            } else if (err.param == 'email') {
                                this.setState({
                                    emailError: err.msg
                                })
                            } else if (err.param == 'password') {
                                this.setState({
                                    passwordError: err.msg
                                })
                            } else if (err.param == 'phoneNumber') {
                                this.setState({
                                    numberError: err.msg
                                })
                            }
                        });
                    } else {
                        Alert.alert(
                            'Alert Message',
                            ' Registration Complete  ',
                            [
                                { text: 'OK', onPress: () => navigate('LoginScreen') },
                            ]
                        );
                    }
                })

                .catch((error) => {
                    console.error('Errors are:', error);
                });

        }

    }
    

    changePwdType = () => {
        
        if (this.state.showPassword) {
          this.setState({
                icEye: 'visibility',
                showPassword: false,
                password: this.state.password
            })
        } else {
            this.setState({
                icEye: 'visibility-off',
                showPassword: true,
                password: this.state.password
            })
        }
        //  this.setState(newState)
    };


    render() {


        return (
            <ImageBackground source={IconsPath.back} style={{ height: '100%', width: '100%' }}>

                <ScrollView>

                    <SafeAreaView style={styles.container}>



                        <Ionicons name="ios-person" size={30} style={{ top: 129, left: -90 }} />
                        <Ionicons name="ios-mail" size={30} style={{ top: 170, left: -90 }} />
                        <Ionicons name="ios-call" size={30} style={{ top: 210, left: -90 }} />
                        <Ionicons name="ios-lock" size={30} style={{ top: 245, left: -90 }} />




                        <TextInput style={styles.input}
                            underlineColorAndroid="transparent"
                            placeholder="Name"
                            placeholderTextColor="black"
                            autoCapitalize="none"
                            disableFullscreenUI={true}
                            value={this.state.name}
                            onEndEditing={() => this.validator('name')}
                            onChangeText={(text) => this.setState({ name: text })}
                        />


                        <Text style={{ top: -5, color: 'red', fontWeight: 'bold' }}>{this.state.nameError}</Text>


                        <TextInput style={styles.input}
                            underlineColorAndroid="transparent"
                            placeholder=" Email-id"
                            placeholderTextColor="black"
                            autoCapitalize="none"
                            keyboardType="email-address"
                            disableFullscreenUI={true}
                            value={this.state.email}
                            onEndEditing={() => this.validator('email')}
                            onChangeText={(text) => this.setState({ email: text })}
                        />
                        <Text style={{ top: -4, color: 'red', fontWeight: 'bold' }}>{this.state.emailError}</Text>


                        <TextInput style={styles.input}
                            underlineColorAndroid="transparent"
                            placeholder=" mobile number"
                            placeholderTextColor="black"
                            autoCapitalize="none"
                            disableFullscreenUI={true}
                            maxLength={10}
                            value={this.state.number}
                            keyboardType='numeric'
                            onEndEditing={() => this.validator('number')}
                            onChangeText={(text) => this.setState({ number: text })}
                        />
                        <Text style={{ top: -4, color: 'red', fontWeight: 'bold' }}>{this.state.numberError}</Text>


                        <TextInput style={styles.input}
                            underlineColorAndroid="transparent"
                            placeholder=" confirm Password"
                            placeholderTextColor="black"
                            secureTextEntry={this.state.showPassword}
                            autoCapitalize="none"
                            value={this.state.password}
                            disableFullscreenUI={true}
                            onEndEditing={() => this.validator('password')}
                            onChangeText={(text) => this.setState({ password: text })}
                        />
                        <MaterialIcons
                            style={styles.icon}
                            name={this.state.icEye}
                            size={30}
                            onPress={this.changePwdType}
                        />
                        <Text style={{ top: -30, color: 'red', fontWeight: 'bold' }}>{this.state.password}</Text>


                        <View style={{ top: -20, backgroundColor: '#3399FF', width: 250, height: 50, borderRadius: 25 }}>
                            <TouchableOpacity style={{ top: 10, left: 80 }} onPress={() => this.register()}>
                                <Text style={{ color: 'white', fontWeight: 'bold', fontSize: 20 }}>Registration</Text>
                            </TouchableOpacity>
                        </View>




                    </SafeAreaView>

                </ScrollView>

            </ImageBackground>

        )


    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
        top: -100
    },
    input: {

        width: 250,
        height: 50,
        fontSize: 16,
        fontWeight: 'bold',
        paddingLeft: 70,
        borderWidth: 1,
        color: 'black',
        backgroundColor: 'white',
        opacity: 0.4,
        borderRadius: 25,
    },
    icon:{
        top:-40,
        left:100
    }
})