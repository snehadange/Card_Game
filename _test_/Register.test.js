import React from 'react';
import Registration from '../App/Screens/Registration';
import renderer from 'react-test-renderer';
import fetch from 'node-fetch'
import fetchMock from 'fetch-mock';

const register_data = {
        "name":"mamata",
        "email":"mamta@gmail.com",
        "password":"mamta123",
        "phoneNumber":9757177475

}

it('works', () => {
    expect(1).toBe(1);
});


//snapshot testing for react Component
test("Register screen", () => {
    const snap = renderer.create(
        <Registration />
    ).toJSON();
    expect(snap).toMatchSnapshot()
})

describe("Register", () => {

    beforeEach(() => {
        fetchMock.reset();
    })
    afterEach(() => {
        fetchMock.restore();
    });

    test("success register", async () => {
        fetchMock.once("http://localhost:8000/src/router/registerUser", {
            status:200,
            body:JSON.stringify("Successfull:"),
            headers:{'Content-Type': 'application/json',}
        })
        const res = await fetch("http://localhost:8000/src/router/registerUser",
        {
            headers:{
                'Content-Type': 'application/json',
                 },
               method:'post',
               body: JSON.stringify(register_data),
        }).then(function(res){
            expect(res.status).toEqual(200)
            return res.json()
        }).then((json)=>{
            console.log(json);
            expect(json).toBe(json)
        })
       
    })


})

