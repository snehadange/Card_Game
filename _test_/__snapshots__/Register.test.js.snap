// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register screen 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "height": "100%",
      "width": "100%",
    }
  }
>
  <Image
    source={1}
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": "100%",
          "width": "100%",
        },
        undefined,
      ]
    }
  />
  <RCTScrollView>
    <View>
      <RCTSafeAreaView
        emulateUnlessSupported={true}
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
            "top": -100,
          }
        }
      >
        <Text />
        <Text />
        <Text />
        <Text />
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          disableFullscreenUI={true}
          onChangeText={[Function]}
          onEndEditing={[Function]}
          placeholder="Name"
          placeholderTextColor="black"
          rejectResponderTermination={true}
          style={
            Object {
              "backgroundColor": "white",
              "borderRadius": 25,
              "borderWidth": 1,
              "color": "black",
              "fontSize": 16,
              "fontWeight": "bold",
              "height": 50,
              "opacity": 0.4,
              "paddingLeft": 70,
              "width": 250,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
        <Text
          style={
            Object {
              "color": "red",
              "fontWeight": "bold",
              "top": -5,
            }
          }
        >
          
        </Text>
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          disableFullscreenUI={true}
          keyboardType="email-address"
          onChangeText={[Function]}
          onEndEditing={[Function]}
          placeholder=" Email-id"
          placeholderTextColor="black"
          rejectResponderTermination={true}
          style={
            Object {
              "backgroundColor": "white",
              "borderRadius": 25,
              "borderWidth": 1,
              "color": "black",
              "fontSize": 16,
              "fontWeight": "bold",
              "height": 50,
              "opacity": 0.4,
              "paddingLeft": 70,
              "width": 250,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
        <Text
          style={
            Object {
              "color": "red",
              "fontWeight": "bold",
              "top": -4,
            }
          }
        >
          
        </Text>
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          disableFullscreenUI={true}
          keyboardType="numeric"
          maxLength={10}
          onChangeText={[Function]}
          onEndEditing={[Function]}
          placeholder=" mobile number"
          placeholderTextColor="black"
          rejectResponderTermination={true}
          style={
            Object {
              "backgroundColor": "white",
              "borderRadius": 25,
              "borderWidth": 1,
              "color": "black",
              "fontSize": 16,
              "fontWeight": "bold",
              "height": 50,
              "opacity": 0.4,
              "paddingLeft": 70,
              "width": 250,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
        <Text
          style={
            Object {
              "color": "red",
              "fontWeight": "bold",
              "top": -4,
            }
          }
        >
          
        </Text>
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          disableFullscreenUI={true}
          onChangeText={[Function]}
          onEndEditing={[Function]}
          placeholder=" confirm Password"
          placeholderTextColor="black"
          rejectResponderTermination={true}
          style={
            Object {
              "backgroundColor": "white",
              "borderRadius": 25,
              "borderWidth": 1,
              "color": "black",
              "fontSize": 16,
              "fontWeight": "bold",
              "height": 50,
              "opacity": 0.4,
              "paddingLeft": 70,
              "width": 250,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
        <Text
          style={
            Object {
              "color": "red",
              "fontWeight": "bold",
              "top": -3,
            }
          }
        >
          
        </Text>
        <View
          style={
            Object {
              "backgroundColor": "#3399FF",
              "borderRadius": 25,
              "height": 50,
              "top": 7,
              "width": 250,
            }
          }
        >
          <RNGestureHandlerButton
            collapsable={false}
            onGestureEvent={[Function]}
            onGestureHandlerEvent={[Function]}
            onGestureHandlerStateChange={[Function]}
            onHandlerStateChange={[Function]}
            rippleColor={0}
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                undefined,
              ]
            }
          >
            <View
              accessible={true}
              style={
                Object {
                  "left": 80,
                  "opacity": 1,
                  "top": 10,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 20,
                    "fontWeight": "bold",
                  }
                }
              >
                Registration
              </Text>
            </View>
          </RNGestureHandlerButton>
        </View>
      </RCTSafeAreaView>
    </View>
  </RCTScrollView>
</View>
`;
